index.js

```diff
@@ -27,18 +28,20 @@ module.exports = (options) => {
             require('postcss-for')(),
             // Add support for @if and @else statements
             require('postcss-conditionals')(),
-            // Add support for nesting
-            require('postcss-nesting')(),
-            // Use `postcss-css-variables` instead of `postcss-custom-properties` because it's more complete
-            // Note that it must be set after the nesting!
-            options.cssVariables && require('postcss-css-variables')(options.cssVariables),
-            // Use CSS next, disabling some features
-            require('postcss-cssnext')({
+            // Use postcss-preset-env, disabling some features
+            require('postcss-preset-env')({
+                stage: 3,
+                insertAfter: {
+                    'nesting-rules': [
+                        // Use `postcss-css-variables` instead of `postcss-custom-properties` because it's more complete
+                        // Note that it must be set after the nesting!
+                        options.cssVariables && require('postcss-css-variables')(options.cssVariables),
```

Comment:
Add `postcss-calc` as a dependency and require it here before `postcss-css-variables` and the `calc` with variables will work.

Beware the options too.